src/lib/perf/perf_counter.cpp

```diff
@@ -262,7 +262,7 @@ perf_end(perf_counter_t handle)
 
 					// maintain mean and variance of the elapsed time in seconds
 					// Knuth/Welford recursive mean and variance of update intervals (via Wikipedia)
-					float dt = elapsed / 1e6f;
+					float dt = elapsed;
```

Comment:
It would be interesting to monitor delta_intvl over a long period of time . I wonder if what you observe is not a large overshoot of the estimated mean, followed by a super slow decay. If so, delta_intvl would be always negative, or at least have negative mean.